src/gameStateMachine.ts

```diff
@@ -0,0 +1,286 @@
+import { createMachine, assign } from "xstate";
+
+interface Artist {
+    id: number;
+    name: string;
+}
+
+interface Lyric {
+    text: string;
+    referentId: number;
+}
+
+interface Lyrics {
+    warped: Lyric[];
+}
+
+interface Song {
+    id: number;
+    title: string;
+    artistName: string;
+    lyrics: Lyrics;
+}
```

Comment:
These are used in the page I'm working on at the moment, once we have this one merged to master and I've pulled it into my feature branch I'll have a look at putting them somewhere shared!

Reply:
Yeah for shared types we can move them into just a big `types.ts` file at the root. Seems to be relatively standard and good enough for us atm.